WorkflowSwiftUI/Sources/ObservableState.swift

```diff
@@ -0,0 +1,181 @@
+import Foundation
+import IdentifiedCollections
+
+/// A type that emits notifications to observers when underlying data changes.
+///
+/// Conforming to this protocol signals to other APIs that the value type supports observation.
+/// However, applying the ``ObservableState`` protocol by itself to a type doesn’t add observation
+/// functionality to the type. Instead, always use the ``ObservableState()`` macro when adding
+/// observation support to a type.
+#if !os(visionOS)
+public protocol ObservableState: Perceptible {
+    var _$id: ObservableStateID { get }
+    mutating func _$willModify()
+}
+#else
```

Comment:
Do we want this else at all? Eg doesn't `Perceptible` vs. `Observable` support different types (structs vs classes) for observation?